internal/net/call/call.go

```diff
@@ -984,8 +996,24 @@ func (c *serverConnection) runHandler(hmap *HandlerMap, id uint64, msg []byte) {
 		}
 	}()
 
+	// Extract trace context and create a new child span to trace the method
+	// call on the server.
+	span := trace.SpanFromContext(ctx) // noop span
+	if sc := readTraceContext(dec); sc != nil {
+		if sc.IsValid() {
```

Comment:
How about checking for error and returning early on error instead of placing the error-handling in the else branch?

Reply:
Done